doc(setting): eventual danger zone setting update (#858)
longhorn/website: content/docs/1.6.0/advanced-resources/deploy/node-selector.md

   >   systemManagedComponentsNodeSelector: "label-key1:label-value1"
   >  ```

### Setting up Node Selector After Longhorn has been installed

> **Warning**:
> * Since all Longhorn components will be restarted, the Longhorn system is unavailable temporarily.

> > * When all Longhorn volumes are detached, the customized setting is immediately applied to the system-managed components (e.g, Instance manager, engine image, CSI driver, etc.).

Author: Updated. Thanks

> * When one or more Longhorn volumes are still attached, the customized setting is applied to the Instance Manager only when no engines and replica instances are running. You are required to reconfigure the setting after detaching the remaining volumes. Alternatively, you can wait for the next setting synchronization, which will occur in an hour.
> * Don't operate the Longhorn system while node selector settings are updated and Longhorn components are being restarted.

1. Prepare

>    * To apply a setting immediately, stop all workloads and detach all Longhorn volumes before applying it.

Reviewer: ```suggestion
   * To ensure that your preferred settings are immediately applied, stop all workloads and detach all Longhorn volumes before configuring the settings.
```

Author: Updated. Thanks

   >   systemManagedComponentsNodeSelector: "label-key1:label-value1"
   >  ```

### Setting up Node Selector After Longhorn has been installed

> **Warning**:
> * Since all Longhorn components will be restarted, the Longhorn system is unavailable temporarily.

> > * When all Longhorn volumes are detached, the customized setting is immediately applied to the system-managed components.

Reviewer 2: Can you briefly describe what components will be applied with the setting.

Author: Added what the components will be applied with the setting.

> > * For Longhorn volumes all detached, the customized setting will be applied to system-managed components immediately.
> > * For one or more volumes in use, the customized setting will be applied to the Instance Manager without engine and replica instances running, and you are required to reconfigure this setting when all volumes are finally detached.

Reviewer: ```suggestion
> * When all Longhorn volumes are detached, the customized settings are immediately applied to the system-managed components.
> * When one or more Longhorn volumes are still attached, the customized settings are applied to the Instance Manager only when no engines and replica instances are running. You might be required to reconfigure the settings after detaching the remaining volumes.
```

> * Don't operate the Longhorn system while node selector settings are updated and Longhorn components are being restarted.

1. Prepare
   * If you are changing node selector in a way so that Longhorn cannot run on some nodes that Longhorn is currently running on,
     you will lose the volume replicas on those nodes.
     Therefore, It is recommended that you evict replicas and disable scheduling for those nodes before changing node selector.

>      See [Evicting Replicas on Disabled Disks or Nodes](../../../nodes-and-volumes/nodes/disks-or-nodes-eviction) for more details about how to do this.

Reviewer 2: This is incorrect anymore. Why wasn't it removed? Please review others as well.

>If you are changing node selector in a way so that Longhorn cannot run on some nodes that Longhorn is currently running on,
     you will lose the volume replicas on those nodes.
     Therefore, It is recommended that you evict replicas and disable scheduling for those nodes before changing node selector.
     See [Evicting Replicas on Disabled Disks or Nodes](../../../nodes-and-volumes/nodes/disks-or-nodes-eviction) for more details about how to do this.

Author: Removed and checked other danger zone settings. Thanks

>    * Suggest stopping all workloads and detaching all Longhorn volumes.

Reviewer: ```suggestion
   * For best results, stop all workloads and detach all Longhorn volumes.
```

Author: Modified. Thanks

>    * For best results, stop all workloads and detach all Longhorn volumes.

Reviewer 2: ```suggestion
   * To apply a setting immediately, stop all workloads and detach all Longhorn volumes before applying it.
```

`best results` is not clear to me, as the result is whether the setting is applied or not. There is no gray area.

Please update others with the same problem.

Author: Modified and all others. Thanks

> **Warning**:
> * Since all Longhorn components will be restarted, the Longhorn system is unavailable temporarily.

> > * Ensure that all Longhorn volumes are detached. When volumes are in use, only Instance Manager is restarted conditionally so you might be required to configure this setting again when all volumes are finally detached.

Reviewer 2: Because the behavior of restarting user-deployed or system-managed components should be the same for some dangerous settings, I would recommend adding the following note next the `Danger Zone` section, so we don't repeat the same explanation.

> For * settings, when all volumes are detached, the setting will be applied immediately, and user-deployed and system-managed components will get restarted. Otherwise, the setting will not be applied until all volumes are detached during the next setting sync-up time. So, you might be required to configure this setting again to apply it immediately when all volumes are finally detached.

Reviewer 2: @mantissahz Please help update the PR, and we should make sure users will understand when the setting will be applied or not.

Author: @innobead,
We only restart Longhorn system-managed components now for users who deploy Longhorn using GitOps tools.
https://github.com/longhorn/longhorn-manager/commit/a02bea80295b67691806226ae03a8d82f0a9eda6

Author: Added the recommended note into the `Danger Zone Setting Configuration` section in `Important note.`

> > * Make sure all Longhorn volumes are `detached`. If there are running Longhorn volumes in the system, this means only the instance manager component will be restarted eventually and you might need to set this setting again when all Longhorn volumes are `detached`.

Reviewer: ```suggestion
> * Ensure that all Longhorn volumes are detached. When volumes are in use, only Instance Manager is restarted so you might be required to configure this setting again when all volumes are finally detached.
```

Author: Modified. Thanks.
